Root/OptimizationDump.cxx

```diff
@@ -377,13 +377,17 @@ EL::StatusCode OptimizationDump :: execute ()
   }
 
   if(!m_inputMET.empty() && !m_inputJets.empty()){
-    m_effectiveMass = VD::Meff_inclusive(in_met, in_jets, in_muons, in_electrons)/1000.;
+    
+    ConstDataVector<xAOD::JetContainer> presel_jets = VD::subset_using_decor(in_jets, VD::decor_vd_pass_Presel, 1);
+    //std::cout << in_jets->size() << " " << presel_jets.size() << std::endl;
+    m_effectiveMass = VD::Meff_inclusive(in_met, presel_jets.asDataVector(), in_muons, in_electrons)/1000.;
```

Comment:
Jets that pass preselection. “signal” is really a misnomer— this is just a state set by SUSYTools. Our own definition of signal is “jets that pass preselection.” This is why I make a subset with this flag.

Best,
Max

On Oct 15, 2015, at 8:53 PM, Giordon Stark <notifications@github.com<mailto:notifications@github.com>> wrote:


In Root/OptimizationDump.cxx<https://github.com/kratsg/TheAccountant/pull/136#discussion_r42202270>:

> @@ -377,13 +377,17 @@ EL::StatusCode OptimizationDump :: execute ()
>    }
>
>    if(!m_inputMET.empty() && !m_inputJets.empty()){
> -    m_effectiveMass = VD::Meff_inclusive(in_met, in_jets, in_muons, in_electrons)/1000.;
> +
> +    ConstDataVector<xAOD::JetContainer> presel_jets = VD::subset_using_decor(in_jets, VD::decor_vd_pass_Presel, 1);
> +    //std::cout << in_jets->size() << " " << presel_jets.size() << std::endl;
> +    m_effectiveMass = VD::Meff_inclusive(in_met, presel_jets.asDataVector(), in_muons, in_electrons)/1000.;


meff needs to be calculated using only jets that pass preselection or signal?

—
Reply to this email directly or view it on GitHub<https://github.com/kratsg/TheAccountant/pull/136/files#r42202270>.